### PR TITLE
Add personalisation link when bad location was set

### PR DESCRIPTION
**Pull request by pfctdayelise**
Detect when ISS gives a 402 error specifically which indicates a bad location. Closes https://trello.com/c/NYgRDR1i/186-unable-to-change-the-location

Added test for this case, and also a test for the case of what we show with other errors, just for completeness.

### CODE REVIEW

File: src/pages/ResultsListPage.js [hunk 1]
```diff
@@ -47,26 +45,22 @@ class ResultsListPage extends BaseResultsPage {
                             <div>Searching...</div>
                     }
                     secondaryText={
-                        this.state.meta ?
+                        this.state.statusCode == 402 ?
```
Comment (danni) [inline src/pages/ResultsListPage.js:45]: `&& this.state.error`. It might have failed to parse.

File: src/pages/ResultsListPage.js [hunk 2]
```diff
@@ -84,6 +78,27 @@ class ResultsListPage extends BaseResultsPage {
         );
     }
 
+    errorMessage(): React.Element {
+        var message = this.state.error;
+        return <p className="errorMessage">{ message }</p>;
```
Comment (danni) [inline src/pages/ResultsListPage.js]: Style: no space between `{ }`

File: src/pages/ResultsListPage.js [hunk 3]
```diff
@@ -84,6 +78,27 @@ class ResultsListPage extends BaseResultsPage {
         );
     }
 
+    errorMessage(): React.Element {
+        var message = this.state.error;
```
Comment (danni) [inline src/pages/ResultsListPage.js:77]: Style: Newline here

File: test/support/mock_iss/server.js [hunk 4]
```diff
@@ -248,6 +248,18 @@ app.get('/api/v3/search/', (req, res) => {
                 services.domesticviolence,
             ],
         });
+    } else if (req.query.q.match(/elasticsearch/)) {
+        res
```
Comment (danni) [inline test/support/mock_iss/server.js:258]: Make this something more explicit. i.e. elasticsearch unavailable.

File: test/support/mock_iss/server.js [hunk 5]
```diff
@@ -248,6 +248,18 @@ app.get('/api/v3/search/', (req, res) => {
...
+            .status(503)
+            .json({
+                error_message: 'The Elasticsearch service is unavailable.',
+            });
+    } else if (req.query.area == "carlt") {
```
Comment (danni) [inline test/support/mock_iss/server.js]: Should be higher in the branches. Else a valid search will override it, and it's effectively always an error.

File: src/pages/ResultsListPage.js (hunk 1 above)
Comment (pfctdayelise) [inline src/pages/ResultsListPage.js:45]: Because the error and statusCode are set simultaneously I think this is ok as-is.

**Comment by danni** [comment]
++

**Pull request merged**